backend/telemetry_core/src/find_location.rs

```diff
@@ -52,28 +50,18 @@ where
     );
 
     // Create a locator with our cache. This is used to obtain locations.
-    let locator = Locator::new(cache);
+    let locator = Arc::new(Locator::new(cache));
 
     // Spawn a loop to handle location requests
     tokio::spawn(async move {
-        // Allow 4 requests at a time. acquiring a token will block while the
-        // number of concurrent location requests is more than this.
-        let semaphore = Arc::new(Semaphore::new(4));
-
         loop {
             while let Ok((id, ip_address)) = rx.recv_async().await {
-                let permit = semaphore.clone().acquire_owned().await.unwrap();
                 let mut response_chan = response_chan.clone();
-                let locator = locator.clone();
+                let locator = Arc::clone(&locator);
 
-                // Once we have acquired our permit, spawn a task to avoid
-                // blocking this loop so that we can handle concurrent requests.
                 tokio::spawn(async move {
-                    let location = locator.locate(ip_address).await;
+                    let location = locator.locate(ip_address);
```

Comment:
More specifically, I think this is where call should be wrapped into blocking task